minidump-processor/src/processor.rs

```diff
@@ -566,6 +553,50 @@ where
     let linux_maps = dump.get_stream::<MinidumpLinuxMaps>().ok();
     let _memory_info = UnifiedMemoryInfoList::new(memory_info_list, linux_maps).unwrap_or_default();
 
+    // Get exception info if it exists.
+    let exception_stream = dump.get_stream::<MinidumpException>().ok();
+    let exception_ref = exception_stream.as_ref();
+
+    let exception_info = exception_ref.map(|exception| {
+        let reason = exception.get_crash_reason(system_info.os, system_info.cpu);
+        let address = exception.get_crash_address(system_info.os, system_info.cpu);
+
+        let (instruction_str, memory_accesses) = exception
+            .context(&dump_system_info, misc_info.as_ref())
+            .map(|context| {
+                let mut memory_list_cursor = MemoryListCursor::new(&memory_list);
+
+                let instruction_str = crate::op_analysis::pretty_print_thread_instruction(
+                    &context,
+                    &mut memory_list_cursor,
+                )
```

Comment:
Hmm so to be clear this code wouldn't do well at reporting something like "hey this JIT code tried to jump to the middle of an instruction", yeah? I know that's not entirely reliable to detect. I'm also not sure if the chunk of memory that minidumps grab includes any bytes before the crash address or if it's just after.

(This is great as is, just thinking out loud.)

Reply:
I think it would work just fine! It will just attempt to start decoding bytes at wherever `rip` is pointing. If that happens to be a  valid instruction, it will pretty-print it even if it's garbage

Reply:
(But if yaxpeax fails to decode it because it's an invalid instruction, we won't pretty-print it)

Reply:
> I'm also not sure if the chunk of memory that minidumps grab includes any bytes before the crash address or if it's just after.

It is typically 256 bytes centered on the instruction pointer. I implemented that in Breakpad for mac and Linux, attempting to match the default behavior from Windows, and remember writing fairly thorough tests for it.